rotkehlchen/api/rest.py

```diff
@@ -1100,12 +1110,16 @@ def add_history_event(self, event: 'HistoryBaseEntry') -> Response:
         result_dict = _wrap_in_ok_result({'identifier': identifier})
         return api_response(result_dict, status_code=HTTPStatus.OK)
 
-    def edit_history_event(self, event: 'HistoryBaseEntry') -> Response:
+    def edit_history_event(self, main_event: 'HistoryBaseEntry', related_events: list['HistoryBaseEntry'] | None) -> Response:  # noqa: E501
         db = DBHistoryEvents(self.rotkehlchen.data.db)
-        result, msg = db.edit_history_event(event)
+        result, msg = db.edit_history_event(main_event)
         if result is False:
             return api_response(wrap_in_fail_result(msg), status_code=HTTPStatus.CONFLICT)
 
+        if related_events is not None and len(related_events) > 0:
+            for event in related_events:
+                db.edit_history_event(event)
```

Comment:
if you check in both methods you edited to:

1. duplicate the same code that you use in the main_event
2. Add a for loop and here for example it doesn't even have proper handling of the events. You shouldn't differienciate between main and related